tests/kernel/mutex/sys_mutex/src/thread_competition.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) 2021 intel, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+#include <sys/mutex.h>
+
+#define HIGH_T1	 0xaaa
+#define HIGH_T2	 0xbbb
+#define LOW_PRO	 0xccc
+
+#define STACKSIZE (512 + CONFIG_TEST_EXTRA_STACKSIZE)
+
+static K_THREAD_STACK_DEFINE(thread_low_stack, STACKSIZE);
+static struct k_thread thread_low_data;
+static K_THREAD_STACK_DEFINE(thread_high_stack1, STACKSIZE);
+static struct k_thread thread_high_data1;
+static K_THREAD_STACK_DEFINE(thread_high_stack2, STACKSIZE);
+static struct k_thread thread_high_data2;
+SYS_MUTEX_DEFINE(mutex);
+static uint32_t flag;
+
+static void low_prio_wait_for_mutex(void *p1, void *p2, void *p3)
+{
+	struct sys_mutex *pmutex = p1;
+
+	sys_mutex_lock(pmutex, K_FOREVER);
+
+	flag = LOW_PRO;
+	/* Keep mutex for a while */
+	k_sleep(K_MSEC(500));
```

Comment:
Half a second is actually a long time in a test suite with hundreds of tests, ideally entire tests should run to completion in a second or two.  Is there a reason for the very long delay?

Reply:
OK, done, thanks.